### ISSUE TITLE
Bugs

### ISSUE THREAD

**Issue by jomibauer** [jomibauer/unity#22]
Add when u find em.  Move to a comment with an explanation of the fix when you fix it

- [ ] The 'x2' turns modifier sprite doesn't turn off in the skirmish preview pane, so if you target someone you'll double attack then target someone who you'll only attack once after that, the lil 'x2' is still there.
- [ ] Sometimes exp doesnt get awarded
- [ ] A player unit cant move the cursor over a square an enemy is in when selecting where to move, but if the cursor can go around them, they're allowed to pass through the enemy square.

**Comment by jomibauer** [comment]
- [x] a weird thing is happening with the 'TURN' model.  I dont think it's getting set properly, and its causing units to not be reset after a new turn starts.  They change color to look active, but they are not able to move again.  I think this may be impacted by whether the unit acts or not, as i notice this happening to units who move, then end the turn. 

Fixed this by clearing the Turn in the Exit Player Phase State.  I think this just resets it but probably needs a lil more testing before we can say its totally fine.

**Comment by jomibauer** [comment]
- [x] I think levels are broken, no stats are increasing on level up 

UnitGrowths and UnitStats were being stored in separate UnitStat objects.  Pay attention to where stats are getting created in the unitParser.

**Comment by jomibauer** [comment]
- [x] Inventories are being weird.  Think we need to make sure they're getting set 'dirty' when parsed like unit stats and other stuff. 

Fixed this by overriding the base Disable method in WeaponSelectionState.  Now the WeaponSelection Options are set to null whenever we disable or switch from the state

**Comment by jomibauer** [comment]
- [x] Selecting a unit that's inactive during the moveTargetState gets a nullRefException 

 just add a null check before doing anything with the selected unit.  Gonna bring up the map menu if the unit is null.

**Comment by jomibauer** [comment]
- [x] Also another one thats a pain to test because IDK what happened, but I had my blue guys attack Gooane and Norridge.  Slimelda hit and killed Gooane, and then Blobby went up and attacked Norridge.  Norridge may have gotten a critical, but it looked to me like Blobby missed, Norridge hit him for 7 damage, then Blobby disappeared and we got a KeyNotFound error.  Weird!! 

I fixed this because unitDied was not getting set back to false at the start of each Skirmish.  If I want a variable to be reset everytime I start a state, I need to do in in the Enable method, not anywhere else.

**Comment by jomibauer** [comment]
- [x] If there are no hits in a Skirmish, the state gets stuck on the SkirmishPlayState.  I don't know why, and this is kind of a pain to test right now since I haven't set up a way to directly control battle outcomes.  I should set up a test scene that lets me test skirmishes with hardcoded rounds so I can control if a unit hits or misses.

fixed, it was an issue with changing the timing of state changes. Encapsulated the stateChange that occurs when there's been no hit inside of a Coroutine, fixed it

**Comment by jomibauer** [comment]
- [x] another issue with targeting units in a 1-2 range, if you're swapping between units, the skirmish range isnt setting properly.

This didnt have to do with swapping.  The distance calculation wasn't using the absolute value of the calculation so if a unit attacked to the left or from below it was considered negative range. fixed by using mathf.abs on the distance calc.

**Comment by jomibauer** [comment]
- [x] Some issue with switching targets with a unit with 1-2 range.  pressing next doesn't always cycle back to the previous unit. - fixing this.  It has to do with the placement of the units.  I think I can try fixing this by ordering the attack range list, but then I may need to change the way Next() and previous work if that fails.

I fixed this by simply increasing the limit on i in the loop inside Previous() by 1.  This still works after some testing, so I think its fine.  What was happening is, when my attackRange was missing some elements because I was attacking something close to the edge of the map, the previous loop wouldnt reach all the way to the square just one element before the current selection.  incresing the iterator limit means that it can reach it before breaking. 

Just a thought, thismay not work if im in a corner and my attack range is further limited.  Something to remember.

**Comment by jomibauer** [comment]
- [x] If a unit dies in a skirmish before its round has played, the round still plays even after it has died.

I think this is fixed.  I just needed to check for death in the OnPlayNextRound method in the skirmishPlayController.

**Comment by jomibauer** [comment]
- [x] Sometimes exp doesnt get awarded

The way i am calculating levelDelta meant that sometimes it was a flat reduction of exp, which combined with other reductions meant that exp was 0 or sometimes negative.  Changed it, but now high-level units are getting too much exp.  I should probably set a hard minimum on some values like killPot and deltaLevel and trigger them when the actual value is negative w/ a ternary.

 -- I tweaked the formula of the experience manager to scale xp a bit better as you get really distant from the other units level.  Still probably needs some testing.

**Comment by jomibauer** [comment]
- [x] High skill gives units a chance to hit over 100

just added a function that makes sure the raw hit/ crit chance is between 100 & 0

**Comment by jomibauer** [comment]
- [x] The 'x2' turns modifier sprite doesn't turn off in the skirmish preview pane, so if you target someone you'll double attack then target someone who you'll only attack once after that, the lil 'x2' is still there.

this was just getting hidden in the SkirmishPreviewPane's Hide method.  I added an else block to where we turn it on in the Show method to turn it off in these cases.
